interfaces/portal/src/app/components/confirmation-dialog/confirmation-dialog.component.ts

```diff
@@ -46,16 +48,31 @@ export class ConfirmationDialogComponent<TMutationData = unknown> {
 
   readonly confirmDialog = viewChild.required<ConfirmDialog>('confirmDialog');
 
+  private event: TrackingEvent | undefined = undefined;
+
   askForConfirmation({
     resetMutation = true,
-  }: { resetMutation?: boolean } = {}) {
+    event = undefined,
```

Comment:
Having this 'callback'-functionality is a nice find.
It does feel a bit 'icky' as somehow I have the idea that "tracking/logging code" should not be _too connected_ to all the other code...

But maybe renaming this to `trackingEvent` might make it clear that this is not to be confused with "a browser/DOM event" or a "Activity/Event from the back-end"...